src/async_client.rs

```diff
@@ -165,8 +171,12 @@ impl RequestDetails {
             body: None,
         }
     }
+    pub async fn send(self, client: &Client) -> Result<Response, Error> {
```

Comment:
Nit: missing new line before new function. Please run `cargo fmt`.

Reply:
> Nit: missing new line before new function. Please run `cargo fmt`.

Fixed this. Interestingly `cargo fmt` doesn't add a new line here though.